### ISSUE TITLE
async Marked extensions don't work properly

### ISSUE THREAD

**Issue by styxpilled** [BearToCode/carta-md#4]
Async Marked extensions need to be awaited, and so currently they don't work.
In `carta-md/$lib/internal/carta`:
```ts
public async render(markdown: string): Promise<string> {
  return new Promise((resolve) => {
    marked.parse(markdown, (e, dirty) => {
      resolve((this.options?.sanitizer && this.options?.sanitizer(dirty)) ?? dirty);
    });
  });
}
```
As in the [marked docs](https://marked.js.org/using_pro#async), this needs to be awaited if someone (me) passes an async extension. I think the same thing stands for the `renderSSR` function but I battled Marked for 5 hours now and I'm tired. I'll submit a PR when I can. A minimal repro of an async marked extension can be found [here](https://github.com/markedjs/marked/issues/2626#issuecomment-1293116640), if you want to get on this before me.

**Comment by BearToCode** [comment]
Hello,
I undestand the issue, it's a little more tricky than simply adding await to the `marked.parse` function because that will break the code highlighting plugin. Luckily, it seems that marked team recently released [marked-highlight](https://www.npmjs.com/package/marked-highlight?activeTab=versions), that will allow async code highlighting and async parsing to work together.
Keep in mind that async extensions will not work in SSR renderings, as it would otherwise require a top level await in svelte components, which are [not yet supported](https://github.com/sveltejs/svelte/issues/5501).
Anyway, I'll try using marked-highlighting and let you know how it goes

**Comment by BearToCode** [comment]
Released new version 1.1.13 with support for async marked extensions

**Comment by styxpilled** [comment]
Thanks! 1.1.13 works flawlessly.
The lack of top level await in Svelte really sucks, I hope that with the internal changes coming in v4 it will get some love.
